lstchain/reco/dl1_to_dl2.py

```diff
@@ -391,20 +392,33 @@ def apply_models(dl1, classifier, reg_energy, reg_disp_vector, custom_config={})
 
     if source_dependent_analysis:
         print("source-dependent analysis is activated")
-        set_source_dependent_parameters(dl2)
+        # To prevent parameters(time_gradient, skewness) from being overwritten in dl2 frame 
+        temp_dl2 = dl2.copy(deep=True)
+        set_source_dependent_parameters(temp_dl2)
+    
         config["regression_features"].append("dist")
         config["classification_features"].append("dist")   
 
     regression_features = config["regression_features"]
     classification_features = config["classification_features"]
-      
+    
+    if source_dependent_analysis:
+        rf_data = temp_dl2
+    else:
+        rf_data = dl2
```

Comment:
Hi.
I am lost here.
What is the purpose of the `temp_dl2` and `rf_data`?

Reply:
If another parameter which will be used for random forest is defined,e.g. 'time_gradient_times_sign', another config file is needed for the regression_features.
But it is unfavorable to overwrite the parameters with the same parameter name.

So here I prepared(copied) the data file only for the random forest(temp_dl2), then the data file for random forest can be chosen depending on the source-dependent/independent analysis.

But there are some other possibilities:
1) save the source-dependent parameters with the different name(e.g. 'time_gradient_times_sign'), and also prepare another parameter for random forest input(like 'time_gradient_times_rf '), then this value depends on the source-dependent/independent.

2) save the source-dependent parameters with the different name(e.g. 'time_gradient_times_sign'), and prepare another config file(or another definition of 'regression_features')

3) like the current one

do you have any ideas...?
 for random forest input